File: 0962-maximum-width-ramp/0962-maximum-width-ramp.java
class Solution {
    public int maxWidthRamp(int[] nums) {
        Stack<Integer> st = new Stack<>();
        int ans = 0;
        for (int i = 0; i < nums.length; i++) {
            if (st.isEmpty() || nums[st.peek()] > nums[i]) {
                st.push(i);
            }
        }
        for (int i = nums.length - 1; i >= 0; i--) {
            while (!st.isEmpty() && nums[st.peek()] <= nums[i]) {
                ans = Math.max(ans, i - st.peek());
                st.pop();
            }
        }
        return ans;
    }
}

File: LeetCode/0962-maximum-width-ramp/0962-maximum-width-ramp.py
class Solution:
    def maxWidthRamp(self, nums: List[int]) -> int:
        st = []
        ans = 0
        for i in range(len(nums)):
            if not st or nums[st[-1]] > nums[i]:
                st.append(i)
        for i in range(len(nums)-1, -1, -1):
            while st and nums[st[-1]] <= nums[i]:
                ans = max(ans, i - st[-1])
                st.pop()
        return ans